tasks/agent.py

```diff
@@ -627,23 +629,43 @@ def omnibus_manifest(
 
 
 @task
-def check_supports_python_version(_, filename, python):
+def check_supports_python_version(_, check_dir, python):
     """
-    Check if a setup.py file states support for a given major Python version.
+    Check if a Python project states support for a given major Python version.
     """
     if python not in ['2', '3']:
         raise Exit("invalid Python version", code=2)
 
-    with open(filename, 'r') as f:
-        tree = ast.parse(f.read(), filename=filename)
+    project_file = os.path.join(check_dir, 'pyproject.toml')
+    setup_file = os.path.join(check_dir, 'setup.py')
+    if os.path.isfile(project_file):
+        with open(project_file, 'r') as f:
+            data = toml.loads(f.read())
 
-    prefix = f'Programming Language :: Python :: {python}'
-    for node in ast.walk(tree):
-        if isinstance(node, ast.keyword) and node.arg == "classifiers":
-            classifiers = ast.literal_eval(node.value)
-            print(any(cls.startswith(prefix) for cls in classifiers), end="")
+        project_metadata = data['project']
+        if 'requires-python' not in project_metadata:
             return
```

Comment:
For the Omnibus code above to make sense, we should always either print `True` or `False` or raise an exception. Here we are not doing either